src/components/carousel/index.tsx

```diff
@@ -1,87 +1,98 @@
-import React, {
-  FC,
-  PropsWithChildren,
-  ReactNode,
-  useCallback,
-  useEffect,
-  useState,
-} from 'react';
+import React, { FC, ReactNode, useCallback, useEffect, useState } from 'react';
 import useEmblaCarousel from 'embla-carousel-react';
-import { useMultiStyleConfig } from '@chakra-ui/react';
+import { useMediaQuery, useMultiStyleConfig } from '@chakra-ui/react';
 
+import ThumbnailPagination from './ThumbnailPagination';
 import Slide from './Slide';
-import NavigationButton, { Direction } from './NavigationButton';
+import NavigationButton from './NavigationButton';
 import Flex from '../flex';
 import Box from '../box';
 
-interface Props {
+type SharedProps = {
   startIndex?: number;
   onSlideClick?: (index: number) => void;
   onSlideSelect?: (index: number) => void;
-  fullScreen?: boolean;
-}
-
-const Carousel: FC<PropsWithChildren<Props>> = ({
-  startIndex = 0,
-  onSlideClick,
-  onSlideSelect,
-  children,
-  fullScreen = false,
-}) => {
-  const { container, slideContainer } = useMultiStyleConfig(
+};
+
+type DefaultProps = {
+  fullScreen?: never;
+  children: ReactNode[];
+} & SharedProps;
+
+type FullScreenSlide = { slide: ReactNode; thumbnail: ReactNode };
+type FullScreenProps = {
+  fullScreen: true;
+  children: Array<FullScreenSlide>;
+} & SharedProps;
+
+type Props = DefaultProps | FullScreenProps;
+
+const Carousel: FC<Props> = (props) => {
+  const { startIndex = 0, onSlideClick, onSlideSelect, fullScreen } = props;
+
+  const hasPagination = !!fullScreen;
+  const numberOfSlides = props.children.length;
+  // 926px is the highest phone
+  const [isMobileLandscape] = useMediaQuery(
```

Comment:
I would avoid introducing a different hardcoded handling here, perhaps I missed something as I don't quite get how you can use the media query as a height.

Reply:
how would I hide the pagination on mobile landscape only without a media-query? do you have another suggestion?

Reply:
and height because: height becomes width on landscape

Reply:
It's less the orientation query that caught me but the hardcoded number, currently for a lot of cases, the `md` breakpoint was used as a threshold for mobile layouts whilst here we use a number slightly below where I would assume it may come from analytics.

The other thing that confuses me is that you are passing a boolean as the height property so the value will either be `true OR  'full' OR 'calc(100% - 124px)'` for the height.

What I am wondering, is if the orientation of the device is the right criteria in general or if we instead could define a minimal height we want to be present in order to render the thumbnails instead.

In general, I would like to avoid having a very specific definition of what "is mobile" within the component and rather use a general approach that is uniform across components and applications

Reply:
as discussed (confirmed everything with Enrique)

we don't use the mobile landscape query but a hardcoded 600px height. This covers small mobile devices that are landscape as well as large desktop with very little height

We decided against a breakpoint query because this would hide it on certain mobile devices in portrait mode